### DISCUSSION TITLE
Meta issue: microsoft-github-policy-service[bot] "This issue has not had any activity in 6 months"

### DISCUSSION THREAD

**Discussion by iRon7** [PowerShell/PowerShell#20821] [General]
### Summary of the new feature / enhancement

Yesterday I got about 30 notifications and today about 90 notifications from the **microsoft-github-policy-service[bot]**
<sup>(Note that more than half of the concerned notifications where redundant)</sup>

> This issue has not had any activity in 6 months, if this is a bug please try to reproduce on the latest version of PowerShell and reopen a new issue and reference this issue if this is still a blocker for you.

I don't think it is appropriate to assume that issues without any "activity" can be automatically closed and therefore I would at least expect the concerned bot to have a more sensitive filter

## Filter
Narrow the Bot filter

### Label
Issues with certain labels should be excluded:
* <kbd>Needs-Triage</kbd> should be removed (by the PowerShell team) so that the author has the acknowledgement that someone from the PowerShell Team is aware.
* <kbd>Up-for-Graps</kbd> is already "*Up-for-Graps*"
* Consider a new label: <kbd>wishlist</kbd> "[*Issue we would like to prioritize, but we can't commit we will get to it yet*](https://github.com/dotnet/runtime/labels?q=wishlist)"

### Inactive
The timestamp of the last comment is not all that defines that the issue inactive
* (emojis) thumbs could be added later to any comment (if placed on the initial issue description, they should a considered with the visibility and priority of the issue)
* Other issues could be linked to the concerned issue at a later state
* Labels could be changed at a later state

### Pester
> reproduce on the latest version of PowerShell

<span />

A possible way to automate this (for both sides) could be a formatted section with an "Automatic Closure Script" pester script in the **Issue: Bug report 🐛** form. Like for #17731 (wishful thinking):

---
### Automatic Closure Script:
```PowerShell
{ [System.Collections.Generic.List[bool]]$True } | Should -Not -Throw
```
---

**Comment by MartinGC94** [comment]
At a minimum they need to fix the bot so it doesn't post 3 comments for every "expired" issue.

Personally I don't understand the reasoning for having the bot do this in the first place. They say it's because there's no value in keeping 3500 issues around because it's hard to manage and they hope that the most important issues will become easier to find by having users reopen each expired issue and linking back to the original.  
That feels like pointless busywork that doesn't respect the users time. Creating a well formulated feature request, or a well documented bug report takes a decent amount of time and on top of that people now have to recreate the issue every 6 months as a reminder that the issue still exists.  
But why? GitHub already allows you to sort by comments and 👍 reactions and while the system isn't perfect, it at least gives the team and community a good starting point for finding potentially high impact issues. How about we fix these seemingly important issues and then if we somehow run out of issues to fix then we can start looking for different metrics, or alternative ways to find important issues?

**Comment by MatejKafka** [comment]
+1

I opened many issues here, keep them up to date with changes, some of them took multiple hours each of my unpaid time to debug, triage and describe. It's disheartening to have most of them closed, especially by a bot, just because I've already described the issue properly and don't have anything to add  except for spamming "still relevant" every 6 months from now on.

Frankly, I might as well not bother opening new issues and donating my time for free, when the issue is just gonna get closed by a bot, since the chance of anyone fixing it before that is very low.

I like PowerShell and so far, I've been happy to contribute, but please, do better than this bullshit.

**Comment by floh96** [comment]
@StevenBucher98 please look into this issue and the linked issues for community feedback regarding the stale bot.
In my opinion, the bot should be reconsidered.

**Comment by dkaszews** [comment]
It's ridiculous, the stupid bot is even closing issues that are waiting on WG response or are already accepted and waiting for PR. Plus it looks to be completely broken, as I have seen multiple threads being closed despite continued discussion. It makes PowerShell look like abandonware with maintainers putting more effort into shutting discussion down than facilitating its evolution.

**Comment by kilasuit** [comment]
across the WG's & the committe this will get some discussion happen that'll eventually find us a much better solution than this, so please bear with us whilst we try find time to discuss and make this a better experience going forward.

**Comment by fMichaleczek** [comment]
Like many others, I no longer want to participate. I feel like I wasted my community time.

**Comment by StevenBucher98** [comment]
We are working on this, hopefully should be fixed by #20758

**Comment by dkaszews** [comment]
@StevenBucher98 This only fixes the "multiple comments" part, not the fact that valid issues are closed because maintainers cannot be bothered to look at them.

**Comment by dkaszews** [comment]
And the fucking bot just closed https://github.com/PowerShell/PowerShell/issues/3316 which has 50+ upvotes and bunch of comments last week trying to keep it open. Please just kill it before it kills the community.

**Comment by kilasuit** [comment]
@dkaszews we are working on rejigging the bot so it works better going forward.

I'll reopen that issue now (if not already done) & please going forward tag @StevenBucher98 @SteveL-MSFT or others like myself that can reopen issues in the issue itself so we can act as quickly as possible, thanks.

**Comment by MartinGC94** [comment]
> we are working on rejigging the bot so it works better going forward.

That's disappointing, I was hoping it would get scrapped completely.  
What would the ideal bot look like to you? Obviously you agree that it shouldn't close issues that other people have commented on recently, but what else is planned for it?

**Comment by floh96** [comment]
@kilasuit if the bot is really necessary I propose the following steps:

1. Find correct settings for the bot 
2. Test the bot on a demo repository
3. Get feedback from the community
4. Reopen all issues in the powershell repository which the misconfigured bot closed and remove the labels and bot comments
5. Activate the bot in the powershell repository
6. Stale issues are getting closed

**Comment by davidtrevor23** [comment]
Just in the last two days, looking for Azure Powershell commands, I stumbled upon half a dozen valid issues or feature requests that were closed down by the bot. It is clearly misconfigured. All topics closed by the bot should be automatically reopened. The bot should only be reactivated when these major issues are addressed.

**Comment by kilasuit** [comment]
### TL:DR 
We need the bot, but it needs configured better to help the WG's, committee & community better work out what to prioritise on for 2024 and beyond of the old issues and hopefully by having done this in this way, as frustrating as it seems, we will get a better chance for that. Closed issues can always be reopened or linked to, so please keep on doing so and please give us some time to get things right, as when we do it will end up working out better for all coming to this repo and other PowerShell team Repos going forward.

Also FYI, I wanted to do something kinda similar to help deal with the issues, but different, which I think would have worked out better overall & perhaps not annoyed so many people in the process too, in that I wanted to gather the WG's & maintainers together to go through and ensure all issues were correctly labelled as opposed to closing them. But this has happened so now we must go forward.

I think we'd have only needed a weeks worth of time, whether that was to a single individual or spread across the maintainers & working group members to go through and correctly re-label all of the issues, which I had asked to try and get the WG's together to do this, however it has been harder to organise than I'd originally thought it would be, but I think it still may happen in future.

Expect to hear more about this in the next community call later today (November 30th 2023)

### The longer version

Managing a repo like this, or any repo really, without some level of automation or labeling system & automation to add/remove labels as needed isn't sensible, practical, or efficent use of the time that is available to go towards the maintainence & future developement of the code contained within & the experience for anyone coming to not just this repo but all other PowerShell organisation repos, to report issues, raise PR's or have discussions going forward. 

Therefore, especially for things which require time lag delays from one action to another, we need to make sure we get the configuration of the bot right, which clearly in this case we didn't, but that doesn't mean we can't make things right going forward, including reopening of currently closed issues, with chance to re-prioritise them going into the new year.

I will say that just like building a functioning script it takes time to get these things right, so please bear with us, which I say as a member of the community & not as a PowerShell repo maintainer, because I am not one, I just have some additional access rights as a member of the Interactive UX WG that allow me opportunity to help out more than others in the community can at this time & can understand why this happened this way, even if it has been frustrating & perhaps could have been better handled.

There were 3.5k open issues prior to this & there wasn't a good clear way for the already well stretched WG's & maintainers to go through and prioritise which issues needed to be reviewed and worked on by the team or be passed to the community, and that also doesn't take into account that the WG's also then need to review PR's too. I personally didn't want to see the way to deal with them pan out as it has but it is what it is and we should look to make it better going forward. The only other way would have been getting the WG's together and drilling through all of the issues in just this repo over a period of time that realistically would have needed all members available for anything upto a full week, which may still happen, but is incredibly hard to organise, this just may be a steppingstone to that point to help prioritise or group issues in future.

Any closed issues can either be asked to be reopened as I mentioned previously, or can be linked to in either a meta issue or follow up issue, which is something that we I see is already happening here and there across this repo. We could however do with managing this a bit better going forward and further enhancing things to make things for the working groups and maintainers easier going forward, which, in time, will happen & will make things better for the community going forward.

Unfortunately, there just isn't enough available time, whether funded or unfunded **and** that has the right level of access in this repo **right now,** especially for those outside the team who often have other things going on, to dedicate the time to properly go through the 3.5k issue backlog that it was and ensure they have been correctly triaged plan and priortise them all, which may also include saying no to some or finding as that really would need the Comittee, maintainers & PS team working group members as well as all the graciously given unfunded time from the community members that are also maintainers or working group members which include people like myself too. 

There is always the question around how & who would/could provide additional funds to help secure time that people like myself & others have put in to help further PowerShell, in a more focused way, whether that be direct to the individuals or put back into other community initatives like kicking off additional PowerShell Community events like the organisation of a mass bug bash like was recently done for PSResourceGet. One thing I know is that those of us that are in the Working Groups work so closely with the team & the community because we want to see and help improve PowerShell for us all. 

There is also the question of whether there is need to expand or add to the working groups with other groups that work closely for management & maintainence of not just this repo but other repos going forward, which I think there is a need for as well as someone that is funded to act between the WG's, & I think should also sit on the committe and that doesn't have lots of other intensive projects or other work or other committments that'll ensure that they can properly focus on this role going forward. 

Also just as an FYI at this point I'm still putting time to this repo after having recently been evicted and am now homeless & in temporary accomodation and have had to deal with moving all my stuff into storage, which leaves me unable to help out anywhere near as much or as easily as I was doing beforehand or would like to going forward, but it is what it is and I'll get through my personal issues, whilst trying my best to keep on helping with this repo & community, where and when I can, which isn't something that I intend to stop fully anytime soon and once I am again more settled I intend to put more time and effort in where I can to help boost this repo and things in the wider community as best as I can, but I hope those reading can empathise with my current situ & know that I am doing as best as I possibly can do right now. 2024 gotta be a better year for me & I think will be a better year for the community too with all the new things coming to us across the wider PowerShell & software development communities.
 
But I will say comments like ` because maintainers cannot be bothered to look at them.` is untrue & unfair, but we do get many non-issues & issues where sometimes comments go way off track (even sometimes from myself) that doesn't always help either.

I am hopeful that from this we will get chance to add some more people added in the new year with rights to help in marking comments as off topic, triage and label issues which will in time I expect (& I think the working group members, maintainers & committee do too) will help reduce frustrations in issues getting stale for no reason or accidently missed or PRs sitting for a long time without being reviewed and either accepted or when required rejected or closed in favour of other PR's quicker going into the future, but I'd also like to see there be a way for the community to provide some way of thanking contributors both in some financial way, like via GitHub Sponsorships and/or with things like badges, Microsoft Points, or official swag going forward, which I am sure many would like to be able to buy some official PowerShell swag going forward too as I know I would.

Hope that all helps and looking forward to chatting to everyone at the community call.

**Comment by kilasuit** [comment]
Converting this to a discussion as this is not really an issue but a discussion for us to continue with going forward
